src/CMakeLists.txt

```diff
@@ -253,40 +254,57 @@ source_group("types" FILES ${types_files})
 source_group("utils" FILES ${utils_files})
 
 #--------------------------------------------------------
-# Lib
-add_library(${src_lib_static_nupiccore_solo} STATIC 
+# Lib core_alg (encoders, SP, TM/TP/BackTM, Anomaly,...) 
+add_library(${src_lib_static_nupiccore_alg} STATIC 
     ${algorithm_files} 
     ${encoders_files} 
-    ${engine_files} 
     ${math_files} 
-    ${ntypes_files} 
-    ${os_files} 
-    ${regions_files} 
     ${types_files} 
     ${utils_files} 
 )
 
 
-target_compile_options( ${src_lib_static_nupiccore_solo} PUBLIC ${INTERNAL_CXX_FLAGS})
-target_compile_definitions(${src_lib_static_nupiccore_solo} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
-target_include_directories(${src_lib_static_nupiccore_solo} PRIVATE 
+target_compile_options( ${src_lib_static_nupiccore_alg} PUBLIC ${INTERNAL_CXX_FLAGS} ${CXX_FLAGS_OPTIMIZED})
+target_compile_definitions(${src_lib_static_nupiccore_alg} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
```

Comment:
I am in the process of changing all of this for shared library.

Reply:
still that split should be available (?)

Reply:
But do we really need it split?
If there was a reason to generate separate static libraries perhaps but I don't see any need.